### PR TITLE
chore: add MasterData Controller tests

### PR DESCRIPTION
**Pull request by mf-isst**
## Description

This PR adds basic test cases for the `backend/masterdata/controller` package.

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/masterdata/controller/MaterialPartnerRelationsControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+
+        // then
+        mockMvc.perform(post("/materialpartnerrelations")
+                .param("ownMaterialNumber", materialNumber)
+                .param("partnerBpnl", bpnl)
+                .param("partnerMaterialNumber", partnerMaterialNumber)
+                .param("partnerSupplies", "true")
+                .param("partnerBuys", "true"))
+            .andExpect(status().is4xxClientError());
+
```
Comment (tom-rm-meyer-ISST) [inline backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/masterdata/controller/MaterialPartnerRelationsControllerTest.java:165]: remove empty lines

File: backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/masterdata/controller/MaterialPartnerRelationsControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+        material.setMaterialPartnerRelations(set);
+        partner.setMaterialPartnerRelations(set);
+
+        MaterialPartnerRelation newMpr2 = new MaterialPartnerRelation(material, partner, partnerMaterialNumber2,
+            true, true);
+        HashSet<MaterialPartnerRelation> set2 = new HashSet<>();
+        set.add(newMpr2);
+        material.setMaterialPartnerRelations(set2);
+        partner.setMaterialPartnerRelations(set2);
+
```
Comment (tom-rm-meyer-ISST) [inline backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/masterdata/controller/MaterialPartnerRelationsControllerTest.java:261]: double empty line

File: backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/masterdata/controller/PartnerControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+    @MockBean
+    private PartnerService partnerService;
+    private final ModelMapper modelMapper = new ModelMapper();
+
+    @Test
+    @WithMockApiKey
+    void createPartnerTest() throws Exception {
+        // given
+        PartnerDto partnerDto = new PartnerDto();
+        partnerDto.setBpnl("123");
```
Comment (tom-rm-meyer-ISST) [inline backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/masterdata/controller/PartnerControllerTest.java:64]: Please use more suffisticated data here and make it global variables.

File: backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/masterdata/controller/PartnerControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+        siteDto.setBpns("testBpns");
+
+
+        Partner existingPartner = new Partner();
+        existingPartner.setBpnl(partnerBpnl);
+        existingPartner.setName("TestName");
+        existingPartner.setEdcUrl("testUrl.com");
+
+        // when
+
```
Comment (tom-rm-meyer-ISST) [inline backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/masterdata/controller/PartnerControllerTest.java:130]: remove empty line, also duplicate in 122

File: backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/masterdata/controller/PartnerControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+        PartnerDto dto1 = modelMapper.map(partner1, PartnerDto.class);
+        PartnerDto dto2 = modelMapper.map(partner2, PartnerDto.class);
+        List<PartnerDto> dtoList = Arrays.asList(dto1, dto2);
+        when(partnerService.findAll()).thenReturn(partnerList);
+
+        // then
+        mockMvc.perform(get("/partners/all")
+                .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(MockMvcResultMatchers.status().isOk())
+            .andExpect(MockMvcResultMatchers.jsonPath("$[0].bpnl").exists())
```
Comment (tom-rm-meyer-ISST) [inline backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/masterdata/controller/PartnerControllerTest.java:188]: nice!

File: backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/masterdata/controller/PartnerControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+    void getOwnSitesWithSitesTest() throws Exception {
+        // given
+        Partner ownPartnerEntity = new Partner();
+        ownPartnerEntity.setBpnl("partnerBpnl");
+        TreeSet<Site> treeSet = new TreeSet<>();
+        Site site1 = new Site();
+        site1.setBpns("testBpns");
+        site1.setName("testName");
+
+
```
Comment (tom-rm-meyer-ISST) [inline backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/masterdata/controller/PartnerControllerTest.java:205]: remove duplicate empty line

File: backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/masterdata/controller/PartnerControllerTest.java [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+        treeSet.add(site1);
+        treeSet.add(site2);
+        ownPartnerEntity.setSites(treeSet);
+
+        // when
+        when(partnerService.getOwnPartnerEntity()).thenReturn(ownPartnerEntity);
+        SiteDto dto1 = modelMapper.map(site1, SiteDto.class);
+        SiteDto dto2 = modelMapper.map(site2, SiteDto.class);
+
+
```
Comment (tom-rm-meyer-ISST) [inline backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/masterdata/controller/PartnerControllerTest.java:219]: same

File: backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/masterdata/controller/PartnerControllerTest.java [hunk 8]
```diff
@@ -0,0 +1,259 @@
...
+    }
+
+    @Test
+    @WithMockApiKey
+    void getOwnSitesWithoutSitesTest() throws Exception {
+        // given
+        Partner ownPartnerEntity = new Partner();
+        ownPartnerEntity.setSites(null);
+
+
```
Comment (tom-rm-meyer-ISST) [inline backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/masterdata/controller/PartnerControllerTest.java:237]: same

File: backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/masterdata/controller/PartnerControllerTest.java [hunk 9]
```diff
@@ -0,0 +1,259 @@
...
+    @Test
+    @WithMockApiKey
+    void createPartnerTest() throws Exception {
+        // given
+        PartnerDto partnerDto = new PartnerDto();
+        partnerDto.setBpnl("123");
+        partnerDto.setName("TestName");
+        partnerDto.setEdcUrl("https://exampleTest.com");
+
+
```
Comment (tom-rm-meyer-ISST) [inline backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/masterdata/controller/PartnerControllerTest.java:68]: duplicate emtpy line

File: backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/masterdata/controller/MaterialControllerTest.java [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+        dto2.setOwnMaterialNumber("testMat2");
+
+
+        // when
+        Material material1 = modelMapper.map(dto1, Material.class);
+        Material material2 = modelMapper.map(dto2, Material.class);
+        List<Material> materialList = Arrays.asList(material1, material2);
+        when(materialService.findAll()).thenReturn(materialList);
+
+
```
Comment (tom-rm-meyer-ISST) [inline backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/masterdata/controller/MaterialControllerTest.java:152]: duplicate empty line

File: backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/masterdata/controller/MaterialControllerTest.java [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+    @MockBean
+    private MaterialService materialService;
+
+    private final ModelMapper modelMapper = new ModelMapper();
+
+    @Test
+    @WithMockApiKey
+    void createMaterialTest() throws Exception {
+        // given
+        MaterialEntityDto materialDto = new MaterialEntityDto();
```
Comment (tom-rm-meyer-ISST) [inline backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/masterdata/controller/MaterialControllerTest.java:65]: please use more meaningful test data. You can make the materialDto / material global in this class as you reuse it multiple times

File: backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/masterdata/controller/MaterialControllerTest.java [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+
+
+@WebMvcTest(MaterialController.class)
+@Import({ SecurityConfig.class, ApiKeyAuthenticationProvider.class })
+public class MaterialControllerTest {
+    @Autowired
```
Comment (tom-rm-meyer-ISST) [inline backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/masterdata/controller/MaterialControllerTest.java:53]: add empty line above @Autowired

**Review by tom-rm-meyer-ISST — CHANGES_REQUESTED** [review]
Great effort, thanks a lot!

Please do following adjustments:
- remove empty lines
- use at some times more meaningful test data (e.g. based on example values)
- make some of your given test data global so that you just write it once (if useful)

File: backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/masterdata/controller/PartnerControllerTest.java (hunk 3 above)
Comment (mf-isst) [reply backend/src/test/java/org/eclipse/tractusx/puris/backend/stock/masterdata/controller/PartnerControllerTest.java]: done as far as possible

**Review by tom-rm-meyer-ISST — APPROVED** [review]
LGTM, thanks a lot!

**Pull request merged**